src/main/java/org/traccar/protocol/LaipacProtocolEncoder.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2015 - 2019 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.protocol;
+
+import org.traccar.Context;
+import org.traccar.Protocol;
+import org.traccar.StringProtocolEncoder;
+import org.traccar.model.Command;
+import org.traccar.helper.Checksum;
+
+public class LaipacProtocolEncoder extends StringProtocolEncoder {
+
+    private final Protocol protocol;
+    private final String defaultDevicePassword;
+
+    public LaipacProtocolEncoder(Protocol protocol) {
+        this.protocol = protocol;
+        defaultDevicePassword = Context.getConfig().getString(
```

Comment:
What about per-device password?

Reply:
That's the job of initDevicePassword call on line 42

Reply:
So, what is the point of this then?

Reply:
Just to set a default password (in the configuration file, can be another place) in case no per-device password is configured using web interface

Reply:
I don't think it's the right way to do with. If you want password to be configurable in the config file, we should probably use lookup attribute in the `initDevicePassword` method.

Reply:
Can it be a "Server" wide attribute?

Reply:
It can be, but if you want to change it, probably create a separate PR. In this one I would keep default password static.

Reply:
Ok. Doing it the way you suggest. Thanks.